### PR TITLE
[#1731] Add Folding documentation to the Eclipse Support website.

### PR DESCRIPTION
**Pull request by miklossy**
Signed-off-by: Tamas Miklossy <miklossy@itemis.de>

### CODE REVIEW

File: xtext-website/documentation/310_eclipse_support.md [hunk 1]
```diff
@@ -482,6 +482,61 @@ public Class<? extends IComparator>
 
 Xtext also provides a quick outline: If you press CTRL-O in an Xtext editor, the outline of the model is shown in a popup window. The quick outline also supports drill-down search with wildcards. To enable the quick outline, you have to put the [QuickOutlineFragment2]({{site.src.xtext_core}}/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/outline/QuickOutlineFragment2.xtend) into your workflow.
 
+## Folding {#folding}
+
+Xtext calculates the editor folding regions based on the grammar out of the box. Althought it comes with good defaults, sometimes they do not satisfy the needs and have to be customized.
```
Comment (cdietrich) [inline xtext-website/documentation/310_eclipse_support.md:487]: -t

File: xtext-website/documentation/310_eclipse_support.md [hunk 2]
```diff
@@ -482,6 +482,61 @@ public Class<? extends IComparator>
...
+
+Considering e.g the Xtext Statemachine example, the framework provides folding capabilities only for the `state` regions:
```
Comment (cdietrich) [inline xtext-website/documentation/310_eclipse_support.md:489]: only nachstellen

File: xtext-website/documentation/310_eclipse_support.md (hunk 1 above)
Comment (miklossy) [reply xtext-website/documentation/310_eclipse_support.md]: done

File: xtext-website/documentation/310_eclipse_support.md (hunk 2 above)
Comment (miklossy) [reply xtext-website/documentation/310_eclipse_support.md:489]: I would leave `only` exactly where it is, otherwise it sounds a little bit weird to me.

Comment (cdietrich) [reply xtext-website/documentation/310_eclipse_support.md:489]: @kthoms @szarnekow what do you think

Comment (szarnekow) [reply xtext-website/documentation/310_eclipse_support.md:489]: You can completely remove `only` from that sentence.

File: xtext-website/documentation/310_eclipse_support.md [hunk 3]
```diff
@@ -482,6 +482,61 @@ public Class<? extends IComparator>
...
 
+## Folding {#folding}
+
+Xtext calculates the editor folding regions based on the grammar out-of-the-box. Althought it comes with good defaults, sometimes they do not satisfy the needs and have to be customized.
+
+Considering e.g the Xtext Statemachine example, the framework provides folding capabilities only for the `state` regions:
+
+![](images/folding_default.png)
+
+In order to make the `events`, `resetEvents` and the `commands` regions foldable, a custom implementation of the [DefaultFoldingRegionProvider]({{site.src.xtext_eclipse}}/org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/folding/DefaultFoldingRegionProvider.java) is necessary:
```
Comment (szarnekow) [inline xtext-website/documentation/310_eclipse_support.md:493]: `In order to make events, resetEvents and commands foldable, too, a custom... `

File: xtext-website/documentation/310_eclipse_support.md (hunk 2 above)
Comment (miklossy) [reply xtext-website/documentation/310_eclipse_support.md]: done

File: xtext-website/documentation/310_eclipse_support.md (hunk 3 above)
Comment (miklossy) [reply xtext-website/documentation/310_eclipse_support.md]: done

**Pull request merged**